### PR TITLE
[ADD] documentation - required knowledge

### PR DESCRIPTION
**Pull request by legalsylvain**
CC : @remytms, @bguillot, @pedrobaeza 
Do you think it is exhaustive ?

### CODE REVIEW

**Comment by pedrobaeza** [comment]
No, you need to know each piece of knowledge in the world. :stuck_out_tongue_closed_eyes: 

Jokes apart, I'll check it and let you know.

**Comment by legalsylvain** [comment]
> No, you need to know each piece of knowledge in the world. 😝

Rare photo of a confirmed OpenUpgrade developer.

image

**Review by remytms — APPROVED** [review]
LGTM.

File: docsource/020_required_knowledge.rst [hunk 1]
```diff
@@ -1,4 +1,30 @@
...
 
-TODO : write here prerequisites.
\ No newline at end of file
+To Use Openupgrade
+------------------
+
+* you should be able to launch an instance of Odoo on your local PC,
+  or on your server, **for each version of your migration**.
+  For example, if you're migrating from version 12.0 to 16.0,
+  you should be able to launch Odoo versions 13.0, 14.0, 15.0 and 16.0.
```
Comment (MiquelRForgeFlow) [inline docsource/020_required_knowledge.rst:12]:
```suggestion
  you should be able to launch Odoo versions 13.0, 14.0, 15.0 and 16.0.
  After the last migration finishes, remember to run update=all without OpenUpgrade to let odoo delete deprecated records.
```

File: docsource/020_required_knowledge.rst [hunk 2]
```diff
@@ -1,4 +1,30 @@
...
+  you'll certainly have to reference numerous open pull requests on OCA/OpenUpgrade,
+  for the modules installed on your instance.
+  In this case, using the `gitaggregate <https://pypi.org/project/git-aggregator>`_
+  tool greatly facilitates the management of these numerous pull requests.
+
+To develop Openupgrade Scripts
+------------------------------
+
+- If you want to develop migration scripts for a given module, you need to have
+  complete functional knowledge of this module.
```
Comment (MiquelRForgeFlow) [inline docsource/020_required_knowledge.rst]:
```suggestion
  complete functional (and technical) knowledge of this module.
```

**Review by OpenCode — APPROVED** [review]

File: docsource/020_required_knowledge.rst [hunk 3]
```diff
@@ -1,4 +1,30 @@
 Required Knowledge
 ==================
 
-TODO : write here prerequisites.
\ No newline at end of file
+To Use Openupgrade
+------------------
+
+* you should be able to launch an instance of Odoo on your local PC,
+  or on your server, **for each version of your migration**.
```
Comment (pedrobaeza) [inline docsource/020_required_knowledge.rst:8]:
```suggestion
  or on your server, **for each version of your migration**.
* You should know how to get `openupgradelib`, obtained from the source repository, installed in your Python environment that is going to run the instance.
* You should know how to invoke the odoo executable, injecting arguments for loading server wide modules and migrations path.
```

File: docsource/020_required_knowledge.rst (hunk 1 above)
Comment (legalsylvain) [reply docsource/020_required_knowledge.rst:12]: thanks for the proposal, but that is not the purpose of this page. this page explain the knowledge you have to have to make a migration, not the way to do a migration.

File: docsource/020_required_knowledge.rst (hunk 2 above)
Comment (legalsylvain) [reply docsource/020_required_knowledge.rst]: Thanks ! applied.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**